### PR TITLE
[Refactor] Move Daily Pokerus Start Generation to its own function in data/pokemon-species

### PR DESCRIPTION
**Pull request by frutescens**
## What are the changes?
This creates a function within data/pokemon-species.ts that allows the game to retrieve the day's list of Pokerus Pokemon. 

## Why am I doing these changes?
I think this will improve clarity of code, but it will also allow the balance team to utilize Pokerus starter generation for other gameplay features.

## What did change?
data/pokemon-species.ts --> Pokemon with Pokerus will be generated through a new function  generatePokemonStarters(scene: BattleScene)
ui/starter-select-ui-handler.ts --> This functionality is removed as a result.

At the moment, there doesn't appear to be any differences in the Pokerus Pokemon generated, but things may change. I do think this is better for the game in the long run. 

### Screenshots/Videos
08-11-2024 PR Pokerus
image

08-11-24 Main Pokerus
Wooper, Beldum, Skelrp --> no filter

## How to test the changes?
Run Pokerogue, look for starters with Pokerogue, and start a new team with them.

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/data/pokemon-species.ts [hunk 1]
```diff
@@ -3317,6 +3317,23 @@ export function getStarterValueFriendshipCap(value: integer): integer {
   }
 }
 
+export function getPokerusStarters(scene: BattleScene): PokemonSpecies[] {
```
Comment (torranx) [inline src/data/pokemon-species.ts:3320]: please add jsdocs

**Review by torranx — APPROVED** [review]

File: src/ui/starter-select-ui-handler.ts [hunk 2]
```diff
@@ -934,6 +902,8 @@ export default class StarterSelectUiHandler extends MessageUiHandler {
       this.starterPreferences = StarterPrefs.load();
     }
     this.moveInfoOverlay.clear(); // clear this when removing a menu; the cancel button doesn't seem to trigger this automatically on controllers
+    this.pokerusSpecies = getPokerusStarters(this.scene as BattleScene);
```
Comment (DayKev) [inline src/ui/starter-select-ui-handler.ts:905]: Shouldn't this always be a `BattleScene`? If it somehow isn't then that means something is very wrong.

File: src/data/pokemon-species.ts [hunk 3]
```diff
@@ -3317,6 +3317,28 @@ export function getStarterValueFriendshipCap(value: integer): integer {
   }
 }
 
+/**
+* Method to get the daily list of starters with Pokerus.
+* @param scene {@linkcode BattleScene} used as part of RNG
+* @returns A list of starters with Pokerus
+*/
+export function getPokerusStarters(scene: BattleScene): PokemonSpecies[] {
+  const pokerusStarters : PokemonSpecies[] = [];
```
Comment (DayKev) [inline src/data/pokemon-species.ts:3326]:
```suggestion
  const pokerusStarters: PokemonSpecies[] = [];
```

File: src/ui/starter-select-ui-handler.ts (hunk 2 above)
Comment (frutescens) [reply src/ui/starter-select-ui-handler.ts:905]: let me re-check but in my recollections, typedocs was being very very silly

Comment (frutescens) [reply src/ui/starter-select-ui-handler.ts:905]: nevermind! whatever was going on - it was fine.

**Review by DayKev — APPROVED** [review]

**Review by KimJeongSun — APPROVED** [review]
LGTM!

**Pull request merged**